server/master/src/main/java/org/apache/accumulo/master/metrics/Metrics2ReplicationMetrics.java

```diff
@@ -58,6 +59,7 @@ protected void snapshot() {
     registry.add(PENDING_FILES, getNumFilesPendingReplication());
     registry.add(NUM_PEERS, getNumConfiguredPeers());
     registry.add(MAX_REPLICATION_THREADS, getMaxReplicationThreads());
+    registry.add(LATENCY, getLatencyInSeconds());
```

Comment:
Ah, crap. It looks like I was confusing what exists in other Metrics libraries as primitives.

Hadoop Metrics2 supports gauges, counters, stats and quantiles. You would want to create `MutableQuantiles` and some `MutableGauges` from the `MetricsRegistry`. You can maintain the min, median, avg, and max as gauges, and then present the percentiles using the quantiles. Essentially, you'd have to track both separately. Like you have here, you could just instrument this class to compute the current metrics instead of doing it "in-band" in `RCRR`.